src/app/pages/home/home.page.ts

```diff
@@ -1,10 +1,42 @@
 import { Component } from '@angular/core';
+import { Platform } from '@ionic/angular';
+import { PushService, PushMessage } from '../../services/push.service';
+import { LocalNotifications } from '@ionic-native/local-notifications/ngx';
+import { NavController } from '@ionic/angular';
+
+declare var document: any;
 
 @Component({
-  selector: 'app-home',
-  templateUrl: 'home.page.html',
-  styleUrls: ['home.page.scss'],
+    selector: 'app-home',
+    templateUrl: 'home.page.html',
+    styleUrls: ['home.page.scss'],
 })
 export class HomePage {
+    private background = false;
+
+    constructor(
+        private localNotifications: LocalNotifications,
+        plt: Platform,
+        push: PushService,
+        public navCtrl: NavController) {
+        // We need to wait for the platform to initialize the plugins
+        plt.ready().then(() => {
```

Comment:
I believe that home page may not be best place to put this. It works, but it is not ideal in terms of showcasing this in code. Home page is one of the many possible places to land in the app. The safest way is to extract this code into the service and call:

`pushService.init()` from `app.component`